DaandB/Week 3/UMLOpdrachtWeek3.cs

```diff
@@ -0,0 +1,110 @@
+﻿﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Enemy : MonoBehaviour {
+
+    public float speed;
+    public GameObject target;
+
+    private void Die() {
+        // Move enemy back to it's spawnpoint (of hoe je het ook wilt gaan doen)
+    }
+
+    private void MoveTowardsPlayer() {
+        // Move the ememy toward the player.
+
+    }
+
+    private void EatPlayer() {
+        // -1 to player lives
+        // if the player's lives are reduced to 0, Player's Die() funcion is called.
+    }
+}
+
+public class Ghost : Enemy {
+
+    private void Respawn() {
+        // Respawns the Ghost at its designated spawn point
+    }
+}
+
+public class BossGhost : Enemy {
+
+    public int health;
+    public float speed;
+    public int pelletsEaten;
+
+    private void PelletCount() {
+        // Counts the amount of pellets eaten by the boss
```

Comment:
This actually counts pellets eaten by the player. and affects the boss thus it being the in the boss class. I understand why this may be confusing and i might do this differently.